pyClient/zeth/joinsplit.py

```diff
@@ -425,7 +327,10 @@ def joinsplit(
             outputs: List[Tuple[ZethAddressPub, EtherValue]],
             v_in: EtherValue,
             v_out: EtherValue,
-            tx_payment: EtherValue) -> contracts.MixResult:
+            tx_payment: EtherValue,
+            compute_h_sig_cb: Optional[ComputeHSigCB] = None,
+            modify_proof_json_cb: Optional[ModifyProofJsonCB] = None
```

Comment:
`modify_proof_json_cb: Optional[ModifyProofJsonCB] = None` do we actually use such callback?

Reply:
This should have been added for the "nullifier + r" attack. 
Because of the non malleability update, we compute `h_sig = sha256(vk || nf_0 || nf_1)`. For the "nullifier + r"  to pass, we need the `h_sig` check to pass. We thus need to update the value of `h_sig` to `h_sig^{rogue} = sha256(vk || nf_{0}^{rogue} || nf_1^{rogue})`
Similarly we need the proof check to pass, so we need to recompute the proof with `h_sig^{rogue}`.

Reply:
My question is about the callback: `modify_proof_json_cb: Optional[ModifyProofJsonCB]`, not the one for h_sig

Reply:
Sorry misread !

Reply:
removed